x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx

```diff
@@ -111,7 +134,7 @@ export const QueryBar = memo<QueryBarComponentProps>(
         dateRangeFrom={dateRangeFrom}
         dateRangeTo={dateRangeTo}
         filters={filters}
-        indexPatterns={indexPatterns as DataView[]}
```

Comment:
I can see that similar casting is used in multiple places in Security Solution, for the same unified SearchBar component from `@kbn/unified-search-plugin/public`

<img width="590" alt="Screenshot 2023-08-31 at 18 05 21" src="https://github.com/elastic/kibana/assets/92328789/e27cc57b-3efa-4bed-93bc-c16d0a2b7a9f">


Can remained castings lead to similar crash as in root cause ticket?

Reply:
In general the castings are bad. I'm not sure if there are other instances where editing a saved filter will lead to this same issue but we should definitely work towards updating the code to remove these castings. This was part of the work I was attempting to fix with this old pr to fix `[useFetchIndex](https://github.com/elastic/kibana/pull/160079) but it's a very big lift to change these castings as a lot of parent components need updating, and then the tests etc..

Reply:
The one casting used in `x-pack/plugins/security_solution/public/common/components/filter_bar/index.tsx`, which is used in EQL rule form. As EQL rule also can have filters, I think it might cause exactly the same crash.

Not sure about another casting, but it feels like we need eventually address it, since it can lead to unexpected issues.